sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -673,8 +680,29 @@ private[joins] class SortMergeJoinScanner(
    */
   private[this] var matchJoinKey: InternalRow = _
   /** Buffered rows from the buffered side of the join. This is empty if there are no matches. */
-  private[this] val bufferedMatches =
+  private[this] val bufferedMatches: AppendOnlyUnsafeRowArray = if (bufferFirstOnly) {
+    new AppendOnlyUnsafeRowArray {
+      var buffer: UnsafeRow = null
+
+      override def clear(): Unit = {
+        buffer = null
+      }
+
+      override def add(row: UnsafeRow): Unit = {
+        assert(buffer == null)
```

Comment:
Are you saying that `ExternalAppendOnlyUnsafeRowArray` will do spill even if we only add one row?

Reply:
No, its threshold parameters do work as expected. Just `ExternalAppendOnlyUnsafeRowArray` looked a bit heavy weight for this case when we want to store only one row. But we can also use `new ExternalAppendOnlyUnsafeRowArray(1, 1)` for this case.

Reply:
How did you test it if `ExternalAppendOnlyUnsafeRowArray` doesn't spill either?